pandas/core/series.py

```diff
@@ -3678,8 +3678,12 @@ def _reduce(self, op, name, axis=0, skipna=True, numeric_only=None,
         if axis is not None:
             self._get_axis_number(axis)
 
-        # dispatch to ExtensionArray interface
-        if isinstance(delegate, ExtensionArray):
+        if isinstance(delegate, Categorical):
```

Comment:
why are u adding a code path here? the original is much more generic ; need to avoid special cases like this
if u need to handle this specially then the place is in the Categirical itself

Reply:
This is because Categorical deviates here from the standard ExtensionArray (see https://github.com/pandas-dev/pandas/issues/25303 for the issue about that). 
I personally find it clearer with this special case, making it explicit that Categorical has a different signature. And after the deprecation period, we can remove this special case.

If you feel strongly about it, it can indeed be handled inside Categorical. But that means that all the other arrays' `_reduce` method needs to be updated as well to handle (=ignore) `numeric_only`, which is also not clean (and the special case here is only temporarily anyway).

Reply:
this is not more clear and leads to future issues
pls move to _reduce

Reply:
not sure why the others need to change at all you’re logic is circular

Reply:
This would cause more changes, since the problem is actually that the argument `numeric_only` is not passed currently. If we would add it for every `ExtensionArray` call we get problems at the other reduction methods. For instance here: https://github.com/pandas-dev/pandas/blob/master/pandas/core/arrays/numpy_.py#L322 (they don't have `numeric_only`, or `**kwargs` in the method definition). 

So we could change the call for every `ExtenensionArray` to:

`return delegate._reduce(name, skipna=skipna, numeric_only=numeric_only, **kwds)`

but then we would need to make sure every child of ExtensionArray supports this and this is currently not the case.

Reply:
@arnov explained it well. I think we don't want to change the EA interface (`_reduce` is an official part of it) just for this back-compat special case we are going to deprecate. Hence, categorical needs to be handled separately here (but again, this is only temporary)

Reply:
disagree numeric_only is likely not going away anytime soon and even so
the EA simply need to accept it (they can ignore it)

Reply:
@jreback this is *not* about the numeric_only in DataFrame/Series reductions that determines for which columns the reduction is calculated. This is another meaning of  the keyword *only* for categorical that determines whether NaNs should be skipped or not. Please read https://github.com/pandas-dev/pandas/issues/25303

So we are not speaking about removing that general use case of `numeric_only`, but only the one in Categorical.min/max.

Reply:
i’ll look closer but am still -1 on any handling as a special case in the Series call
the point is that pass on kwargs; EA can ignore or not as required

Reply:
OK, to take a step back: @jreback do you agree that in the *long term* we can deprecate this `numeric_only` keyword for Categorical.min/max? 
(it's the only EA that now uses it, while the others all use `skipna` for the same thing)